tapiriik/services/BeginnerTriathlete/beginnertriathlete.py

```diff
@@ -0,0 +1,438 @@
+        ".gpx": _DeviceFileTypes.GPX,
+        ".tcx": _DeviceFileTypes.TCX,
+        ".fit": _DeviceFileTypes.FIT
+    }
+
+    SupportedActivities = list(_workoutTypeMappings.values())
+
+    def WebInit(self):
+        self.UserAuthorizationURL = WEB_ROOT + reverse("auth_simple", kwargs={"service": self.ID})
+
+    # Exchange username & password for a UserToken and store it in ExtendedAuthorization if the user has elected to
+    # remember login details.
+    def Authorize(self, username, password):
+        session = self._prepare_request()
+        requestParameters = {"username": username,  "password": password}
+        user_resp = session.get(self._loginUrlRoot, params=requestParameters)
+
+        if user_resp.status_code != 200:
+            raise APIException("Login error")
+
+        response = user_resp.json()
+
+        if response["LoginResponseCode"] == 3:
+            from tapiriik.auth.credential_storage import CredentialStore
+            member_id = int(response["MemberId"])
+            token = response["UserToken"]
+
+            # Grab the user's timezone. The information retrieved from DownloadActivityList is technically TZ naive,
+            # but the user has a timezone in their settings that we can use to translate.
+            # I think it is OK to grab this value once during authorization, but we could cache it and read it
+            # at the beginning of each synchronization if it causes problems.
+            timezone = self._getTimeZone(token)
+            return member_id, {}, {"UserToken": CredentialStore.Encrypt(token), "TimeZone": timezone}
+
+        if response["LoginResponseCode"] == 0:
+            raise APIException("Invalid API key")
+
+        # Incorrect username or password
+        if response["LoginResponseCode"] == -3 or response["LoginResponseCode"] == -2 or response["LoginResponseCode"] == -1:
+            raise APIException(
+                "Invalid login - Bad username or password",
+                block=True,
+                user_exception=UserException(UserExceptionType.Authorization, intervention_required=True))
+
+        # Account is inactive or locked out - Rarely would happen
+        if response["LoginResponseCode"] == 1 or response["LoginResponseCode"] == 2:
+            raise APIException(
+                "Invalid login - Account is inactive",
+                block=True,
+                user_exception=UserException(UserExceptionType.Authorization, intervention_required=True))
+
+        # Something extra unusual has happened
+        raise APIException(
+            "Invalid login - Unknown error",
+            block=True,
+            user_exception=UserException(UserExceptionType.Authorization, intervention_required=True))
+
+    # Get an activity summary over a date range
+    def DownloadActivityList(self, serviceRecord, exhaustive=False):
+        activities = []
+
+        if exhaustive:
+            # "Exhaustive" currently goes back one year
```

Comment:
Is there a way to actually fetch all activities? As it is, activities more than a year ago on a user's BT account will get duplicated.